File: week-1-c/section/hello/hello.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func main() {
	name, err := getString("What's your name? ")
	if err != nil {
		fmt.Println(err)
	}

	fmt.Printf("Hello, %s!\n", name)
}

func getString(prompt string) (string, error) {
	reader := bufio.NewReader(os.Stdin)

	fmt.Print(prompt)
	input, err := reader.ReadString('\n')
	if err != nil {
		return "", fmt.Errorf("error reading input: %v", err)
	}

	return strings.TrimSpace(input), nil
}

File: week-1-c/problem-sets/me/hello.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

char *get_string(const char *prompt);

int main(void)
{
  char *name = get_string("What's your name? ");
  if (name == NULL)
  {
    fprintf(stderr, "Failed to get input.\n");
    return 1;
  }

  printf("Hello, %s\n", name);

  free(name);
  return 0;
}

char *get_string(const char *prompt)
{
  const int STRING_SIZE = 255 * sizeof(char);
  printf("%s", prompt);

  char *str = malloc(STRING_SIZE);
  if (str == NULL)
  {
    fprintf(stderr, "Memory allocation failed.\n");
    return NULL;
  }

  if (fgets(str, STRING_SIZE, stdin) == NULL)
  {
    free(str);
    return NULL;
  }

  str[strcspn(str, "\n")] = '\0';

  return str;
}
